### PR TITLE
862 add canary guidance to site

### PR DESCRIPTION
**Pull request by GCHQ-Developer-741**
## Summary of the changes
Created a new canary section on the components page, which includes guidance and code examples for the canary date-input, as well as an index page which includes a link to the storybook.

## Related issue
#862 

## Checklist
- [x] I have [manually accessibility tested](https://design.sis.gov.uk/accessibility/testing/manual-testing) any changes, if relevant.
- [x] I have raised this pull request against the [develop branch](https://github.com/mi6/ic-design-system/tree/develop)

### CODE REVIEW

File: src/content/structured/components/canary/date-input/guidance.mdx [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+## Date formatting
+
+Use the date input ‘dateFormat’ property to change the default input format between DD/MM/YYYY, MM/DD/YYYY and YYYY/MM/DD. Use differing labels on the helper text and within a date input, when catering for regional input preferences.
+
+## Interaction behaviour
+
+Use the ‘disablePast’ property when using date input to gather information about future dates, such as hotel bookings.
+
+Use the ‘ disableFuture’ property when using a date input to gather information about days in the past, such as birthdays.
```
Comment (gd2910) [inline src/content/structured/components/canary/date-input/guidance.mdx:113]:
```suggestion
Use the ‘disableFuture’ property when using a date input to gather information about days in the past, such as birthdays.
```

File: src/content/structured/components/canary/date-input/guidance.mdx [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    imageSrc={figure7}
+    imageAlt="A date input that is larger than the button component it is next to."
+    caption="Don’t mix different sized components."
+  />
+</DoubleDoDontCaution>
+
+### Width
+
+Ensure the width of the date input allows enough space for the full length of a person’s potential input.
+
```
Comment (gd2910) [inline src/content/structured/components/canary/date-input/guidance.mdx:145]: One for design, are both images meant to be the correct usage of the width of the date-input?
Seems strange having two 'correct' but differing images.

File: src/content/structured/components/canary/date-input/code.mdx [hunk 3]
```diff
@@ -0,0 +1,556 @@
...
+<ic-date-input
+  label="When would you like to collect your coffee?"
+  format="YYYY/MM/DD"
+></ic-date-input>
+<script>
+  const firstDateInput = document.querySelector("ic-date-input");
+  firstDateInput.addEventListener("icChange", function ({ detail }) {
+    console.log(detail.value);
+  });
+</script>`,
```
Comment (gd2910) [inline src/content/structured/components/canary/date-input/code.mdx:47]:
```suggestion
    snippet: `<ic-date-input label="When would you like to collect your coffee?"></ic-date-input>
<ic-date-input label="When would you like to collect your coffee?" format="MM/DD/YYYY"></ic-date-input>
<ic-date-input label="When would you like to collect your coffee?" format="YYYY/MM/DD"></ic-date-input>
<script>
  const firstDateInput = document.querySelector("ic-date-input");
  firstDateInput.addEventListener("icChange", function ({ detail }) {
    console.log(detail.value);
  });
</script>`,
```

File: src/content/structured/components/canary/date-input/code.mdx [hunk 4]
```diff
@@ -0,0 +1,556 @@
...
+</ComponentPreview>
+
+#### Zulu datetime
+
+export const snippetsZulu = [
+  {
+    language: "Web components",
+    snippet: `<ic-date-input
+  label="When would you like to collect your coffee?"
+></ic-date-input>
```
Comment (gd2910) [inline src/content/structured/components/canary/date-input/code.mdx:159]:
```suggestion
    snippet: `<ic-date-input label="When would you like to collect your coffee?"></ic-date-input>
```

File: src/content/structured/components/canary/date-input/code.mdx [hunk 5]
```diff
@@ -0,0 +1,556 @@
...
+</ComponentPreview>
+
+#### String with hyphens
+
+export const snippetsHyphens = [
+  {
+    language: "Web components",
+    snippet: `<ic-date-input
+  label="When would you like to collect your coffee?"
+></ic-date-input>
```
Comment (gd2910) [inline src/content/structured/components/canary/date-input/code.mdx:188]:
```suggestion
    snippet: `<ic-date-input label="When would you like to collect your coffee?"></ic-date-input>
```

File: src/content/structured/components/canary/date-input/code.mdx [hunk 6]
```diff
@@ -0,0 +1,556 @@
...
+</ComponentPreview>
+
+#### Epoch date
+
+export const snippetsEpoch = [
+  {
+    language: "Web components",
+    snippet: `<ic-date-input
+  label="When would you like to collect your coffee?"
+></ic-date-input>
```
Comment (gd2910) [inline src/content/structured/components/canary/date-input/code.mdx:217]:
```suggestion
    snippet: `<ic-date-input label="When would you like to collect your coffee?"></ic-date-input>
```

File: src/content/structured/components/canary/date-input/code.mdx [hunk 7]
```diff
@@ -0,0 +1,556 @@
...
+
+### Required
+
+export const snippetsRequired = [
+  {
+    language: "Web components",
+    snippet: `<ic-date-input
+  label="When would you like to collect your coffee?"
+  required="true"
+></ic-date-input>`,
```
Comment (gd2910) [inline src/content/structured/components/canary/date-input/code.mdx:247]:
```suggestion
    snippet: `<ic-date-input label="When would you like to collect your coffee?" required="true"></ic-date-input>`,
```

File: src/content/structured/components/canary/date-input/code.mdx [hunk 8]
```diff
@@ -0,0 +1,556 @@
...
+
+### Disabled
+
+export const snippetsDisabled = [
+  {
+    language: "Web components",
+    snippet: `<ic-date-input
+  label="When would you like to collect your coffee?"
+  disabled="true"
+></ic-date-input>`,
```
Comment (gd2910) [inline src/content/structured/components/canary/date-input/code.mdx:267]:
```suggestion
    snippet: `<ic-date-input label="When would you like to collect your coffee?" disabled="true"></ic-date-input>`,
```

File: src/content/structured/components/canary/date-input/code.mdx [hunk 9]
```diff
@@ -0,0 +1,556 @@
...
+  label="When would you like to collect your coffee?"
+  size="small"
+></ic-date-input>
+<ic-date-input
+  label="When would you like to collect your coffee?"
+></ic-date-input>
+<ic-date-input
+  label="When would you like to collect your coffee?"
+  size="large"
+></ic-date-input>`,
```
Comment (gd2910) [inline src/content/structured/components/canary/date-input/code.mdx:294]:
```suggestion
    snippet: `<ic-date-input label="When would you like to collect your coffee?" size="small"></ic-date-input>
<ic-date-input label="When would you like to collect your coffee?"></ic-date-input>
<ic-date-input label="When would you like to collect your coffee?" size="large"></ic-date-input>`,
```

File: src/content/structured/components/canary/date-input/code.mdx [hunk 10]
```diff
@@ -0,0 +1,556 @@
...
+
+### Custom helper text
+
+export const snippetsCustomHelperText = [
+  {
+    language: "Web components",
+    snippet: `<ic-date-input
+  label="When would you like to collect your coffee?"
+  helper-text="We will have your order ready for you on this date"
+></ic-date-input>`,
```
Comment (gd2910) [inline src/content/structured/components/canary/date-input/code.mdx:332]:
```suggestion
    snippet: `<ic-date-input label="When would you like to collect your coffee?" helper-text="We will have your order ready for you on this date"></ic-date-input>`,
```

File: src/content/structured/components/canary/date-input/code.mdx [hunk 11]
```diff
@@ -0,0 +1,556 @@
...
+<ic-date-input
+  label="When would you like to collect your coffee?"
+  validation-status="warning"
+  validation-text="Please be aware that there may be a coffee shortage on that day."
+></ic-date-input>
+<ic-date-input
+  label="When would you like to collect your coffee?"
+  validation-status="success"
+  validation-text="Your coffee will be available for you to collect on this day."
+></ic-date-input>`,
```
Comment (gd2910) [inline src/content/structured/components/canary/date-input/code.mdx:369]:
```suggestion
    snippet: `<ic-date-input label="When would you like to collect your coffee?" validation-status="error" validation-text="There is a coffee shortage forecast for that day. Please choose a different date."></ic-date-input>
<ic-date-input label="When would you like to collect your coffee?" validation-status="warning" validation-text="Please be aware that there may be a coffee shortage on that day."></ic-date-input>
<ic-date-input label="When would you like to collect your coffee?" validation-status="success" validation-text="Your coffee will be available for you to collect on this day."></ic-date-input>`,
```

File: src/content/structured/components/canary/date-input/code.mdx [hunk 12]
```diff
@@ -0,0 +1,556 @@
...
+
+### Disabled past dates
+
+export const snippetsDisabledPastDates = [
+  {
+    language: "Web components",
+    snippet: `<ic-date-input
+  label="When would you like to collect your coffee?"
+  disable-past="true"
+></ic-date-input>`,
```
Comment (gd2910) [inline src/content/structured/components/canary/date-input/code.mdx:425]:
```suggestion
    snippet: `<ic-date-input label="When would you like to collect your coffee?" disable-past="true"></ic-date-input>`,
```

File: src/content/structured/components/canary/date-input/code.mdx [hunk 13]
```diff
@@ -0,0 +1,556 @@
...
+
+### Disabled future dates
+
+export const snippetsDisabledFutureDates = [
+  {
+    language: "Web components",
+    snippet: `<ic-date-input
+  label="When would you like to collect your coffee?"
+  disable-future="true"
+></ic-date-input>`,
```
Comment (gd2910) [inline src/content/structured/components/canary/date-input/code.mdx:451]:
```suggestion
    snippet: `<ic-date-input label="When would you like to collect your coffee?" disable-future="true"></ic-date-input>`,
```

File: src/content/structured/components/canary/date-input/code.mdx [hunk 14]
```diff
@@ -0,0 +1,556 @@
...
+### Min and max
+
+export const snippetsMinMax = [
+  {
+    language: "Web components",
+    snippet: `<ic-date-input
+  label="When would you like to collect your coffee?"
+  min="02/01/2023"
+  max="20-07-2023"
+></ic-date-input>`,
```
Comment (gd2910) [inline src/content/structured/components/canary/date-input/code.mdx:478]:
```suggestion
    snippet: `<ic-date-input label="When would you like to collect your coffee?" min="02/01/2023" max="20-07-2023"></ic-date-input>`,
```

File: src/content/structured/components/canary/date-input/code.mdx [hunk 15]
```diff
@@ -0,0 +1,556 @@
...
+  ></ic-date-input>
+  <ic-button type="submit" value="Submit"></ic-button>
+  <ic-button type="reset" value="Reset"></ic-button>
+  <script>
+    document.querySelector("form").addEventListener("submit", (ev) => {
+      ev.preventDefault();
+      console.log(ev);
+    });
+  </script>
+</form>`,
```
Comment (gd2910) [inline src/content/structured/components/canary/date-input/code.mdx:516]:
```suggestion
  <ic-date-input label="When would you like to collect your coffee?" value="13/03/1951"></ic-date-input>
  <ic-button type="submit" value="Submit"></ic-button>
  <ic-button type="reset" value="Reset"></ic-button>
  <script>
    document.querySelector("form").addEventListener("submit", (ev) => {
      ev.preventDefault();
      console.log(ev);
    });
  </script>
</form>`,
```

File: src/content/structured/components/canary/date-input/accessibility.mdx [hunk 16]
```diff
@@ -0,0 +1,100 @@
...
+      path: "/components/canary/date-input/accessibility",
+    },
+  ]
+---
+
+## Easy to use for everyone
+
+### Keyboard navigation
+
+The day, month, and year values of a date input focus from left to right with focus moving between each date element.
```
Comment (gd2910) [inline src/content/structured/components/canary/date-input/accessibility.mdx:27]:
```suggestion
The day, month and year values of a date input focus from left to right with focus moving between each date element.
```

File: src/content/structured/components/canary/date-input/code.mdx [hunk 17]
```diff
@@ -0,0 +1,556 @@
...
+  onChange={({ detail }) => console.log(detail.value)}
+/>
+<IcDateInput
+  label="When would you like to collect your coffee?"
+  dateFormat="MM/DD/YYYY"
+/>
+<IcDateInput 
+  label="When would you like to collect your coffee?"
+  dateFormat="YYYY/MM/DD"
+/>`,
```
Comment (gd2910) [inline src/content/structured/components/canary/date-input/code.mdx:62]:
```suggestion
    snippet: `<IcDateInput label="When would you like to collect your coffee?" onChange={({ detail }) => console.log(detail.value)}/>
<IcDateInput label="When would you like to collect your coffee?" dateFormat="MM/DD/YYYY"/>
<IcDateInput  label="When would you like to collect your coffee?" dateFormat="YYYY/MM/DD"/>`,
```

File: src/content/structured/components/canary/date-input/code.mdx [hunk 18]
```diff
@@ -0,0 +1,556 @@
...
+  label="When would you like to collect your coffee?"
+  value="21/01/2001"
+></ic-date-input>`,
+  },
+  {
+    language: "React",
+    snippet: `<IcDateInput
+  label="When would you like to collect your coffee?"
+  value="21/01/2001"
+/>`,
```
Comment (gd2910) [inline src/content/structured/components/canary/date-input/code.mdx:112]:
```suggestion
    snippet: `<IcDateInput label="When would you like to collect your coffee?" value="21/01/2001"/>`,
```

File: src/content/structured/components/canary/date-input/code.mdx [hunk 19]
```diff
@@ -0,0 +1,556 @@
...
+  const dateInputDate = document.querySelector("ic-date-input");
+  dateInputDate.value = new Date("28 November 2021");
+</script>`,
+  },
+  {
+    language: "React",
+    snippet: `<IcDateInput
+  label="When would you like to collect your coffee?"
+  value={new Date("28 November 2021")}
+/>`,
```
Comment (gd2910) [inline src/content/structured/components/canary/date-input/code.mdx:141]:
```suggestion
    snippet: `<IcDateInput label="When would you like to collect your coffee?" value={new Date("28 November 2021")}/>`,
```

File: src/content/structured/components/canary/date-input/code.mdx [hunk 20]
```diff
@@ -0,0 +1,556 @@
...
+  const dateInputDate = document.querySelector("ic-date-input");
+  dateInputDate.value = new Date("1 May 1990").toISOString();
+</script>`,
+  },
+  {
+    language: "React",
+    snippet: `<IcDateInput
+  label="When would you like to collect your coffee?"
+  value={new Date("1 May 1990").toISOString()}
+/>`,
```
Comment (gd2910) [inline src/content/structured/components/canary/date-input/code.mdx:170]:
```suggestion
    snippet: `<IcDateInput label="When would you like to collect your coffee?" value={new Date("1 May 1990").toISOString()}/>`,
```

File: src/content/structured/components/canary/date-input/code.mdx [hunk 21]
```diff
@@ -0,0 +1,556 @@
...
+  const dateInputDate = document.querySelector("ic-date-input");
+  dateInputDate.value = "25-05-2005";
+</script>`,
+  },
+  {
+    language: "React",
+    snippet: `<IcDateInput
+  label="When would you like to collect your coffee?"
+  value="25-05-2005"
+/>`,
```
Comment (gd2910) [inline src/content/structured/components/canary/date-input/code.mdx:199]:
```suggestion
    snippet: `<IcDateInput label="When would you like to collect your coffee?" value="25-05-2005"/>`,
```

File: src/content/structured/components/canary/date-input/code.mdx [hunk 22]
```diff
@@ -0,0 +1,556 @@
...
+  const dateInputDate = document.querySelector("ic-date-input");
+  dateInputString.value = Date.now();
+</script>`,
+  },
+  {
+    language: "React",
+    snippet: `<IcDateInput
+  label="When would you like to collect your coffee?"
+  value={Date.now()}
+/>`,
```
Comment (gd2910) [inline src/content/structured/components/canary/date-input/code.mdx:228]:
```suggestion
    snippet: `<IcDateInput label="When would you like to collect your coffee?" value={Date.now()}/>`,
```

File: src/content/structured/components/canary/date-input/code.mdx [hunk 23]
```diff
@@ -0,0 +1,556 @@
...
+  label="When would you like to collect your coffee?"
+  helper-text="We will have your order ready for you on this date"
+></ic-date-input>`,
+  },
+  {
+    language: "React",
+    snippet: `<IcDateInput
+  label="When would you like to collect your coffee?"
+  helperText="We will have your order ready for you on this date"
+/>`,
```
Comment (gd2910) [inline src/content/structured/components/canary/date-input/code.mdx:339]:
```suggestion
    snippet: `<IcDateInput label="When would you like to collect your coffee?" helperText="We will have your order ready for you on this date"/>`,
```

File: src/content/structured/components/canary/date-input/code.mdx [hunk 24]
```diff
@@ -0,0 +1,556 @@
...
+<IcDateInput
+  label="When would you like to collect your coffee?"
+  validationStatus="warning"
+  validationText="Please be aware that there may be a coffee shortage on that day."
+/>
+<IcDateInput
+  label="When would you like to collect your coffee?"
+  validationStatus="success"
+  validationText="Your coffee will be available for you to collect on this day."
+/>`,
```
Comment (gd2910) [inline src/content/structured/components/canary/date-input/code.mdx:387]:
```suggestion
    snippet: `<IcDateInput label="When would you like to collect your coffee?" validationStatus="error" validationText="There is a coffee shortage forecast for that day. Please choose a different date." />
<IcDateInput label="When would you like to collect your coffee?" validationStatus="warning" validationText="Please be aware that there may be a coffee shortage on that day." />
<IcDateInput label="When would you like to collect your coffee?" validationStatus="success" validationText="Your coffee will be available for you to collect on this day." />`,
```

File: src/content/structured/components/canary/date-input/code.mdx [hunk 25]
```diff
@@ -0,0 +1,556 @@
...
+  label="When would you like to collect your coffee?"
+  disable-past="true"
+></ic-date-input>`,
+  },
+  {
+    language: "React",
+    snippet: `<IcDateInput
+  label="When would you like to collect your coffee?"
+  disablePast
+/>`,
```
Comment (gd2910) [inline src/content/structured/components/canary/date-input/code.mdx:432]:
```suggestion
    snippet: `<IcDateInput label="When would you like to collect your coffee?" disablePast />`,
```

File: src/content/structured/components/canary/date-input/code.mdx [hunk 26]
```diff
@@ -0,0 +1,556 @@
...
+  label="When would you like to collect your coffee?"
+  disable-future="true"
+></ic-date-input>`,
+  },
+  {
+    language: "React",
+    snippet: `<IcDateInput
+  label="When would you like to collect your coffee?"
+  disableFuture
+/>`,
```
Comment (gd2910) [inline src/content/structured/components/canary/date-input/code.mdx:458]:
```suggestion
    snippet: `<IcDateInput label="When would you like to collect your coffee?" disableFuture />`,
```

File: src/content/structured/components/canary/date-input/code.mdx [hunk 27]
```diff
@@ -0,0 +1,556 @@
...
+  max="20-07-2023"
+></ic-date-input>`,
+  },
+  {
+    language: "React",
+    snippet: `<IcDateInput
+  label="When would you like to collect your coffee?"
+  min="02/01/2023"
+  max="20-07-2023"
+/>`,
```
Comment (gd2910) [inline src/content/structured/components/canary/date-input/code.mdx:486]:
```suggestion
    snippet: `<IcDateInput label="When would you like to collect your coffee?" min="02/01/2023" max="20-07-2023" />`,
```

File: src/content/structured/components/canary/date-input/code.mdx [hunk 28]
```diff
@@ -0,0 +1,556 @@
...
+return (
+  <form onSubmit={formSubmit}>
+    <IcDateInput
+      label="When would you like to collect your coffee?"
+      value="13/03/1951"
+    />
+    <IcButton type="submit">Submit</IcButton>
+    <IcButton type="reset">Reset</IcButton>
+  </form>
+);`,
```
Comment (gd2910) [inline src/content/structured/components/canary/date-input/code.mdx:533]:
```suggestion
    snippet: `const formSubmit = (ev) => {
  ev.preventDefault();
  console.log(ev);
};
return (
  <form onSubmit={formSubmit}>
    <IcDateInput label="When would you like to collect your coffee?" value="13/03/1951" />
    <IcButton type="submit">Submit</IcButton>
    <IcButton type="reset">Reset</IcButton>
  </form>
);`,
```

**Comment by gd2910** [comment]
Tested with canary-docs import and all is well!

File: src/content/structured/components/canary/date-input/code.mdx (hunk 11 above)
Comment (GCHQ-Developer-741) [reply src/content/structured/components/canary/date-input/code.mdx:369]: Personally think the original is easier to read when displayed in the code preview

**Comment by ad9242** [comment]
there seem to be quite a few date-pickers in the images, which I wouldn't really expect on the date-input page. in particular the "when not to use" image is confusing as it says to use 3 separate fields, but then shows a date picker

**Comment by jd239** [comment]
Some of the guidance or component will need to be updated, for example:
- There is guidance on potentially changing the placeholder text but this is not possible as it uses 3 'micro' inputs.
- A screenshot shows an example where all the placeholders are not visible with a cursor present. When selecting within an input, the other two inputs will still display their placeholder text if not populated
- By default, the helper text is 'use format <format>', but our images show 'Enter in format <format>'
- There is no prop to hide the helperText so we may need to update placeholder text 'do not use' images. A workaround is that the user adds an empty string to the helperText prop but it would not remove the element.

**Comment by GCHQ-Developer-741** [comment]
> Some of the guidance or component will need to be updated:
> 
> * There is guidance on potentially changing the placeholder text but this is not possible as it uses 3 'micro' inputs.
> * A screenshot shows an example where all the placeholders are not visible with a cursor present. When selecting within an input, the other two inputs will still display their placeholder text if not populated
> * By default, the helper text is 'use format ', but our images show 'Enter in format '
> * There is no prop to hide the helperText so we may need to update placeholder text 'do not use' images. A workaround is that the user adds an empty string to the helperText prop but it would not remove the element.

Have made amendments based on these. Screenshots and guidance has been updated. For the last bullet point, I would argue that it is still valid since the ability to add an empty string means that users could avoid displaying a helper text, and this guidance highlights why that is bad practise.

**Review by gd2910 — APPROVED** [review]

**Comment by jd239** [comment]
There seems to be an issue with the padding on `.focus-indicator`. If you view the component within the Storybook, the padding is correct but not on the guidance site. It looks like there is an issue with the cascading on the `focus-indicator` class. It wasn't there when I previously looked at the PR so it may be the Gatsby build doing something with the CSS.
image

**Comment by GCHQ-Developer-299** [comment]
> There seems to be an issue with the padding on `.focus-indicator`. If you view the component within the Storybook, the padding is correct but not on the guidance site. It looks like there is an issue with the cascading on the `focus-indicator` class. It wasn't there when I previously looked at the PR so it may be the Gatsby build doing something with the CSS.
> 
> image

It's going to need more research to find out why Gatsby is importing the CSS in the wrong order (although there are many [threads about it online](https://github.com/gatsbyjs/gatsby/issues/9733) so we're going to make the .focus-indicator rule in ic-date-input.css more specific. Will fix it on fix/ic-date-input-focus-indicator-fix

**Comment by GCHQ-Developer-299** [comment]
PR created to address the styling issue: https://github.com/mi6/ic-ui-kit/pull/1787

**Review by jd239 — APPROVED** [review]

**Review by GCHQ-Developer-465 — APPROVED** [review]

**Pull request merged**